### PR TITLE
Enhancement: Collect code coverage on GitHub Actions

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] collects code coverage on GitHub Actions

Follows #27.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -121,3 +121,46 @@ jobs:
...
+          - 7.2
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          coverage: xdebug
```
Comment (localheinz) [inline .github/workflows/build.yaml:151]: Installing `xdebug` as code coverage driver here.

💁‍♂️ For reference, see:

- https://github.com/shivammathur/setup-php#xdebug

Comment (nikolaposa) [reply .github/workflows/build.yaml:151]: A-ha! 👍

**Review by nikolaposa — APPROVED** [review]
🍰

**Comment by nikolaposa** [comment]
Scrutinizer seems stuck: https://scrutinizer-ci.com/g/nikolaposa/version/inspections/433c1dee-0e2d-4b9b-8fb8-b0814b3a5975. 🤔

**Comment by nikolaposa** [comment]
Hm, it's not stuck anymore but it failed: https://scrutinizer-ci.com/g/nikolaposa/version/inspections/433c1dee-0e2d-4b9b-8fb8-b0814b3a5975/log. @localheinz Have some idea?

**Comment by localheinz** [comment]
@nikolaposa 

I am taking a 👀!

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -121,3 +121,55 @@ jobs:
...
+        if: github.event_name == 'push'
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.sha }}
+
+      - name: Checkout head commit
+        if: github.event_name == 'pull_request'
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (localheinz) [inline .github/workflows/build.yaml:146]: Looks like Scrutinizer accepts the code coverage report (see https://github.com/nikolaposa/version/pull/31/checks?check_run_id=950559775), but cannot attribute it to the commit hash.

`actions/checkout` checks out a theoretical merge commit for pull requests. This commit obviously is not known to Scrutinizer.

By conditionally checking out

- the commit on `push` events (for building pushes to `master`, whether direct or indirect through merges)
- the actual head commit for `pull_request` events

we can work around the issue.

Here, we access the [`github` context](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context), which allows inspecting event payloads!

💁‍♂️ For reference, see:

- https://github.com/actions/checkout#Checkout-pull-request-HEAD-commit-instead-of-merge-commit
- https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#contexts
- https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context
- https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push
- https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#push
- https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request
- https://docs.github.com/en/developers/webhooks-and-events/webhook-events-and-payloads#pull_request

**Comment by localheinz** [comment]
@nikolaposa 

Looks like Scrutinizer is still running, let's see if it works.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (nikolaposa) [reply .github/workflows/build.yaml:146]: Wow, that is an amazing find, well done. 👏

**Comment by localheinz** [comment]
Thank you, @nikolaposa!

**Pull request merged**